pkg/snmp/traps/config.go

```diff
@@ -104,13 +105,17 @@ func (c *Config) Addr() string {
 }
 
 // BuildSNMPParams returns a valid GoSNMP params structure from configuration.
-func (c *Config) BuildSNMPParams() (*gosnmp.GoSNMP, error) {
+func (c *Config) BuildSNMPParams(logger log.Component) (*gosnmp.GoSNMP, error) {
+	var snmpLogger gosnmp.Logger
+	if logger != nil {
+		snmpLogger = gosnmp.NewLogger(&trapLogger{logger: logger})
+	}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Is it safe to continue the program if `snmpLogger` is nil ?

Reply:
Is it expected for `logger log.Component` to be nil? Shall we stop the program when logger is nil?

Reply:
It shouldn't be nil. I've changed this to just create the logger, which matches the existing behavior.